frontend/src/services/graphQL.ts

```diff
@@ -32,14 +35,57 @@ export async function graphQLRequest(query: String, variables: ILookup<any> = {}
   return await axios.request(request)
 }
 
+export async function graphQLRequestWithErrorHandling(query: String, variables: ILookup<any> = {}, parse : any = undefined) {
+  try {
+    const result = await graphQLRequest(
+      query,
+      variables
+    )
+    const errors = graphQLGetErrors(result)
+    if(parse) {
+      await parse(result)
+    }
+    return errors ? 'ERROR' : result
+  } catch (error) {
+    await apiError(error)
+  }
+}
+
 export function graphQLGetErrors(response: AxiosResponse | 'ERROR' | void, silent?: boolean) {
   if (!response || response === 'ERROR') return
   const errors: undefined | { message: string }[] = response?.data?.errors
 
   if (errors) {
     errors.forEach(error => console.warn('graphQL error:', error))
     if (!silent) store.dispatch.ui.set({ errorMessage: 'GraphQL: ' + errors[0].message })
+  } else {
+    // console.log('No errors, setting count to 0')
+    errorCount = 0 //Set error count back to 0, no errors
   }
 
   return errors
 }
+
+export async function apiError(error: unknown) {
+  const { ui, auth } = store.dispatch
+  console.error('API ERROR:', error)
+  console.trace()
+
+  if (axios.isAxiosError(error)) {
+    if (error.response?.status === 401 || error.response?.status === 403) {
+      console.log('Incrementing error count: ', errorCount)
+      errorCount = errorCount + 1
+      auth.checkSession({ bypassCache: true })
+      if( errorCount > 1000 ) {
+        auth.signOut()
+      }
+      sleep(1000 * errorCount)
```

Comment:
Does sleep need an await? Otherwise could just use setTimeout
```suggestion
     await sleep(1000 * errorCount)
```